setup.py

```diff
@@ -325,14 +288,20 @@ def build_extension_cmake(self, ext):
 
         # We derive the cmake build directory from the output directory, but put it in
         # a sibling directory to avoid polluting the final package
-        cmake_build_dir = os.path.abspath(self.build_lib.replace("lib.", "cmake."))
+        cmake_build_dir = os.path.abspath(os.path.join(self.build_lib.replace("lib.", "cmake."), ext.name))
```

Comment:
It would be great if we could use the equivalent utilities from the `pathlib` library

Reply:
I took a stab at doing this in places where the precise sequence of operations was better captured in pathlib:
```
-HERE = os.path.dirname(os.path.abspath(__file__))
+HERE = Path(__file__).resolve().parent

 DEBUG_COMPILE = "DD_COMPILE_DEBUG" in os.environ

@@ -283,12 +284,12 @@ class CMakeBuild(build_ext):

     def build_extension_cmake(self, ext):
         # Define the build and output directories
-        output_dir = os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.name)))
-        extension_basename = os.path.basename(self.get_ext_fullpath(ext.name))
+        output_dir = Path(self.get_ext_fullpath(ext.name)).parent.resolve()
+        extension_basename = Path(self.get_ext_fullpath(ext.name)).name

         # We derive the cmake build directory from the output directory, but put it in
         # a sibling directory to avoid polluting the final package
-        cmake_build_dir = os.path.abspath(os.path.join(self.build_lib.replace("lib.", "cmake."), ext.name))
+        cmake_build_dir = Path(self.build_lib.replace("lib.", "cmake."), ext.name).resolve()
```

Is this what you were thinking, or do you think we should replace the `os.path.join()` constructs with pathlib as well, now that we're using it anyway (in other words, standardize on a single filesystem path management abstraction)?